File: Problems/Week_1/3.22_스택(10828)/임용식/10828.cs
using System;
using System.Text;

namespace AlgoCSharp
{
    class Program
    {
        static void Main(string[] args)
        {
            int commandCount;
            commandCount = Convert.ToInt32(Console.ReadLine());
            StringBuilder sb = new StringBuilder();
            int[] stack = new int[commandCount];
            int ptr = 0;

            while (commandCount-- > 0)
            {
                string[] temp = Console.ReadLine().Split();
                string command = temp[0].ToLower();
                int data = 0;
                if (temp.Length > 1)
                {
                    data = Convert.ToInt32(temp[1]);
                }
                if (command == "push")
                {
                    stack[ptr++] = data;
                }
                else if (command == "pop")
                {
                    if (ptr > 0)
                    {
                        sb.Append(stack[--ptr] + "\n");
                    }
                    else
                    {
                        sb.Append(-1 + "\n");
                    }
                }
                else if (command == "size")
                {
                    sb.Append(ptr + "\n");
                }
                else if (command == "empty")
                {
                    sb.Append((ptr != 0 ? 0 : 1) + "\n");
                }
                else if (command == "top")
                {
                    if (ptr == 0)
                    {
                        sb.Append(-1 + "\n");
                    }
                    else
                    {
                        sb.Append(stack[ptr - 1] + "\n");
                    }
                }
            }
            Console.WriteLine(sb);
        }
    }
}

File: Problems/Week_1/3.22_스택(10828)/임용식/10828.c
#pragma warning(disable:4996)
#include <stdio.h>
#include <string.h>

typedef struct _Stack
{
	int container[10000];
	int ptr;
}Stack;



void Push(Stack*, const int*);
void Pop(Stack*);
void Size(const Stack* const);
void Empty(const Stack* const);
void Top(const Stack* const);

int main()
{
	Stack st = { {0},0 };
	char command[16]= "";
	int count;
	scanf("%d", &count);
	while (count--)
	{
		scanf("%s", command);

		if (strcmp(command, "push") == 0)
		{
			int data;
			scanf("%d", &data);
			Push(&st, &data);
		}
		else if (strcmp(command, "pop") == 0)
		{
			Pop(&st);
		}
		else if (strcmp(command, "size") == 0)
		{
			Size(&st);
		}
		else if (strcmp(command, "empty") == 0)
		{
			Empty(&st);
		}
		else if (strcmp(command, "top") == 0)
		{
			Top(&st);
		}
	}

	return 0;
}

void Push(Stack* st, const int* data)
{
	st->container[st->ptr++] = *data;
}

void Pop(Stack* st)
{
	if (st->ptr == 0)
	{
		printf("-1\n");
		return;
	}
	printf("%d\n", st->container[--(st->ptr)]);
}

void Size(const Stack* const st)
{
	printf("%d\n", st->ptr);
}

void Empty(const Stack* const st)
{
	printf("%d\n", st->ptr == 0 ? 1 : 0);
}

void Top(const Stack* const st)
{
	if (st->ptr == 0)
	{
		printf("-1\n");
		return;
	}
	printf("%d\n", st->container[(st->ptr) - 1]);
}